api/server/sdk/credentials.go

```diff
@@ -589,3 +589,194 @@ func validateAndDeleteIfInvalid(ctx context.Context, s *CredentialServer, uuid s
 
 	return nil
 }
+
+// Update method modifies previously created credentials
+func (s *CredentialServer) Update(
+	ctx context.Context,
+	req *api.SdkCredentialUpdateRequest,
+) (*api.SdkCredentialUpdateResponse, error) {
+	if s.driver(ctx) == nil {
+		return nil, status.Error(codes.Unavailable, "Resource has not been initialized")
+	}
+
+	if len(req.GetCredentialId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply credentialId")
+	} else if aws := req.UpdateReq.GetAwsCredential(); aws != nil {
+		return s.awsUpdate(ctx, req.UpdateReq, aws, req.GetCredentialId())
+	} else if azure := req.UpdateReq.GetAzureCredential(); azure != nil {
+		return s.azureUpdate(ctx, req.UpdateReq, azure, req.GetCredentialId())
+	} else if google := req.UpdateReq.GetGoogleCredential(); google != nil {
+		return s.googleUpdate(ctx, req.UpdateReq, google, req.GetCredentialId())
+	}
+	return nil, status.Error(codes.InvalidArgument, "Unknown credential type")
+}
+
+func (s *CredentialServer) awsUpdate(
+	ctx context.Context,
+	req *api.SdkCredentialCreateRequest,
+	aws *api.SdkAwsCredentialRequest,
+	credId string,
+) (*api.SdkCredentialUpdateResponse, error) {
+
+	params := make(map[string]string)
+
+	params[api.OptCredType] = "s3"
+	params[api.OptCredName] = req.GetName()
+	// Users dont have to provide there if they dont want to change this
+	if len(req.GetEncryptionKey()) != 0 {
+		params[api.OptCredEncrKey] = req.GetEncryptionKey()
+	}
+	if len(req.GetBucket()) != 0 {
+		params[api.OptCredBucket] = req.GetBucket()
+	}
+	if len(aws.GetRegion()) != 0 {
+		params[api.OptCredRegion] = aws.GetRegion()
+	}
+	if len(aws.GetEndpoint()) != 0 {
+		params[api.OptCredEndpoint] = aws.GetEndpoint()
+	}
+	if len(aws.GetAccessKey()) != 0 {
```

Comment:
should we put additional checks before calling driver update for missing access/secrets ?

Reply:
It would be good for server to handle those. Hence not included here.

Reply:
Isn't this server itself ? I see the checks on google but not for aws and azure

Reply:
What I meant is combination of these fields for an update the implementation can decide to check on in whatever way they want. Here we dont need to check as not all of these fields are mandatory for update.